src/lib/database/store.ts

```diff
@@ -295,6 +295,39 @@ const initializer = immer<Database>((set, get) => ({
     })
   },
 
+  updateClimateSettingSchedule(params) {
+    const target = get().climate_setting_schedules.find(
+      (climate_setting_schedule) =>
+        climate_setting_schedule.climate_setting_schedule_id ===
+        params.climate_setting_schedule_id
+    )
+    if (target == null) {
+      throw new Error(
+        "Could not find climate_setting_schedule with climate_setting_schedule_id"
+      )
+    }
+
+    const updated: ClimateSettingSchedule = { ...target, ...params } as any
```

Comment:
We have this linter rule actually:

/Users/michaelanderjaska/fake-seam-connect/src/lib/database/store.ts
  310:21  error  Always prefer const x: T = { ... }  @typescript-eslint/consistent-type-assertions
  
In the past I've always used `as`, so I don't know if this is a good one to keep or not

Reply:
[37d0662](https://github.com/seamapi/fake-seam-connect/pull/56/commits/37d066283cd3cdd898969e30c6d4128cf599ff21)

Reply:
@Andrewjeska Check out, e.g., https://stackoverflow.com/a/54685761

TLDR; There is a very important difference. The `as` is overriding the type system (generally undesired since it can lead to runtime errors, etc.) where the colon is helpful annotation.

Also https://www.totaltypescript.com/tutorials/beginners-typescript/beginner-s-typescript-section/assigning-types-to-variables